monai/data/png_saver.py

```diff
@@ -0,0 +1,125 @@
+
+    @staticmethod
+    def _create_file_basename(postfix, input_file_name, folder_path, data_root_dir=""):
+        """
+        Utility function to create the path to the output file based on the input
+        filename (extension is added by lib level writer before writing the file)
+
+        Args:
+            postfix (str): output name's postfix
+            input_file_name (str): path to the input image file
+            folder_path (str): path for the output file
+            data_root_dir (str): if not empty, it specifies the beginning parts of the input file's
+                absolute path. This is used to compute `input_file_rel_path`, the relative path to the file from
+                `data_root_dir` to preserve folder structure when saving in case there are files in different
+                folders with the same file names.
+        """
+
+        # get the filename and directory
+        filedir, filename = os.path.split(input_file_name)
+
+        # jettison the extension to have just filename
+        filename, ext = os.path.splitext(filename)
+        while ext != "":
+            filename, ext = os.path.splitext(filename)
+
+        # use data_root_dir to find relative path to file
+        filedir_rel_path = ""
+        if data_root_dir:
+            filedir_rel_path = os.path.relpath(filedir, data_root_dir)
+
+        # sub-folder path will be original name without the extension
+        subfolder_path = os.path.join(folder_path, filedir_rel_path, filename)
+        if not os.path.exists(subfolder_path):
+            os.makedirs(subfolder_path)
+
+        # add the sub-folder plus the postfix name to become the file basename in the output path
+        return os.path.join(subfolder_path, filename + "_" + postfix)
+
+    def save(self, data, meta_data=None):
+        """
+        Save data into a png file.
+        The metadata could optionally have the following keys:
+
+            - ``'filename_or_obj'`` -- for output file name creation, corresponding to filename or object.
+            - ``'spatial_shape'`` -- for data output shape.
+
+        If meta_data is None, use the default index from 0 to save data instead.
+
+        args:
+            data (Tensor or ndarray): target data content that to be saved as a png format file.
+                Assuming the data shape are spatial dimensions.
+            meta_data (dict): the meta data information corresponding to the data.
+
+        See Also
+            :py:meth:`monai.data.png_writer.write_png`
+        """
+        filename = meta_data['filename_or_obj'] if meta_data else str(self._data_index)
+        self._data_index += 1
+        spatial_shape = meta_data.get('spatial_shape', None) if meta_data else None
+
+        if torch.is_tensor(data):
+            data = data.detach().cpu().numpy()
+        filename = self._create_file_basename(self.output_postfix, filename, self.output_dir)
+        filename = '{}{}'.format(filename, self.output_ext)
```

Comment:
**Important thing to note here are:** 
- 1 channel is gray is skimage, so while reading it would return (h,w) rather than (h,w,1)
- 3 channel is rgb, so while reading it would return (h,w,3). 
- 2 channel will not work. 
Let me know your views on this.

Reply:
This `moveaxis` to change the model output data, not reading from image.
This PNG saver is mainly used to save model output into files for further usage or analysis.
Because model output is always `channel_first` and needs to change to `channel_last` to save into file.